syntax = "proto3";

package ram_generator;

option go_package = "github.com/KalashnikovProjects/RamGenerator/protos";

message GenerateStartPromptRequest {
  string user_prompt = 1;
}

message GenerateHybridPromptRequest {
  string user_prompt = 1;
  repeated string ram_images = 2;   // base64
}

message ImagePrompt {
  string prompt = 1;
}

message GenerateImageRequest {
  string prompt = 1;
  string style = 2;
}

message GenerateImageResponse {
  string ram_image = 1;           // base64
}


service RamGenerator {
  rpc GenerateStartPrompt(GenerateStartPromptRequest) returns (ImagePrompt);
  rpc GenerateHybridPrompt(GenerateHybridPromptRequest) returns (ImagePrompt);
  rpc GenerateImage(GenerateImageRequest) returns (GenerateImageResponse);
}